packages/server/tests/acceptance/htsPrecompile.spec.ts

```diff
@@ -0,0 +1,95 @@
+/*-
+ *
+ * Hedera JSON RPC Relay
+ *
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// external resources
+import { solidity } from 'ethereum-waffle';
+import chai, { expect } from 'chai';
+
+chai.use(solidity);
+
+import { AliasAccount } from '../clients/servicesClient';
+import { ethers, BigNumber } from 'ethers';
+import ERC20MockJson from '../contracts/ERC20Mock.json';
+import BaseHTSJson from '../contracts/BaseHTS.json';
+
+
+describe('HTS Precompile Acceptance Tests', async function() {
+  this.timeout(240 * 1000); // 240 seconds
+  const { servicesNode, relay } = global;
+
+  const accounts: AliasAccount[] = [];
+  let baseHTSContract;
+  let HTSTokenContract;
+
+  before(async () => {
+    accounts[0] = await servicesNode.createAliasAccount(30, relay.provider);
+    accounts[1] = await servicesNode.createAliasAccount(30, relay.provider);
+    accounts[2] = await servicesNode.createAliasAccount(30, relay.provider);
+  });
+
+  it('should deploy a BaseHTS contract', async function() {
+    const baseHTSFactory = new ethers.ContractFactory(BaseHTSJson.abi, BaseHTSJson.bytecode, accounts[0].wallet);
+    const baseHTS = await baseHTSFactory.deploy();
+    const { contractAddress } = await baseHTS.deployTransaction.wait();
+    expect(contractAddress).to.not.be.null;
+
+    baseHTSContract = new ethers.Contract(contractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    expect(baseHTSContract).to.not.be.null;
+    expect(baseHTSContract.address).to.equal(contractAddress);
+  });
+
+  it('should create a HTS token', async function() {
+    const tx = await baseHTSContract.createToken(accounts[0].wallet.address, {
+      value: ethers.BigNumber.from('20000000000000000000'),
+      gasLimit: 1000000
+    });
+    const { tokenAddress } = (await tx.wait()).events.filter(e => e.event = 'CreatedToken')[0].args;
+
+    expect(tokenAddress).to.not.be.null;
+
+    HTSTokenContract = new ethers.Contract(tokenAddress, ERC20MockJson.abi, accounts[0].wallet);
```

Comment:
Q: does this need to be done in the `before()`?

Reply:
As long as they are async calls we can't initialize them before `describe` and in the `describe` body. So I think that the best place to create these accounts is in `before` hook.

Reply:
The accounts is fine, i'm referring to the Contract deployment.
That should be in the before().
Turn this into a normal function (not a test) and call it in before()

Same logic is used in `erc20.spec.ts` and `rpc.spec.ts` i.e. create main contract in before/beforeAll.
In this way future tests are assured it's present and no tests are run if it fails

Reply:
Oh, got it. :pray: